src/components/encoding-pane/field-customizer.tsx

```diff
@@ -12,35 +14,141 @@ export interface FieldCustomizerProps extends ActionHandler<SpecEncodingAction>
   fieldDef: ShelfFieldDef;
 }
 
+export interface CustomProp {
+  prop: string;
+  nestedProp: string;
+}
 
-export class FieldCustomizerBase extends React.PureComponent<FieldCustomizerProps, {}> {
+const POSITION_FIELD_QUANTITATIVE_INDEX = {
+  'Common': [
+    {
+      prop: 'scale',
+      nestedProp: 'type'
+    },
+    {
+      prop: 'axis',
+      nestedProp: 'title'
+    },
+    {
+      prop: 'stack',
+      nestedProp: undefined
+    }
+  ],
+  'Scale': [
+    {
+      prop: 'scale',
+      nestedProp: 'type'
+    }
+  ],
+  'Axis': [
+    {
+      prop: 'axis',
+      nestedProp: 'orient'
+    },
+    {
+      prop: 'axis',
+      nestedProp: 'title'
+    }
+  ]
+};
+
+const POSITION_FIELD_NOMINAL_INDEX = {
+  'Scale': [
+    {
+      prop: 'scale',
+      nestedProp: 'type'
+    }
+  ],
+  'Axis': [
+    {
+      prop: 'axis',
+      nestedProp: 'orient'
+    },
+    {
+      prop: 'axis',
+      nestedProp: 'title'
+    }
+  ]
+};
+
+const POSITION_FIELD_TEMPORAL_INDEX = {
```

Comment:
Refactor to separate index. Looks identical to `POSITION_FIELD_NOMINAL_INDEX` currently, but in the future can have different properties

Reply:
If it looks identical right now, don't separate as it's hard to read. Plus, if they differ in the future, they will share common parts that should be declared once.